### PR TITLE
refactor(android): migrate DashMediaSource to Kotlin

### PR DESCRIPTION
**Pull request by seyedmostafahasani**
## Summary
Migrate `DashMediaSource` to Kotlin

### Motivation
Migrate the codebase from Java to Kotlin

## Test plan

### CODE REVIEW

File: android/src/main/java/androidx/media3/exoplayer/dash/DashMediaSource.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import androidx.media3.common.MediaItem
+import androidx.media3.datasource.DataSource
+import androidx.media3.exoplayer.drm.DrmSessionManagerProvider
+import androidx.media3.exoplayer.source.MediaSource
+import androidx.media3.exoplayer.upstream.LoadErrorHandlingPolicy
+
+class DashMediaSource {
+    class Factory(private val chunkSourceFactory: DefaultDashChunkSource.Factory, private val dataSourceFactory: DataSource.Factory) : MediaSource.Factory {
+        override fun setDrmSessionManagerProvider(drmSessionManagerProvider: DrmSessionManagerProvider): MediaSource.Factory {
+            TODO("Not yet implemented")
```
Comment (freeboub) [inline android/src/main/java/androidx/media3/exoplayer/dash/DashMediaSource.kt]: Todos are not needed in this file also as this is a stub file

**Review by freeboub — CHANGES_REQUESTED** [review]

File: android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+import androidx.media3.exoplayer.dash.manifest.DashManifest;
+import androidx.media3.exoplayer.trackselection.ExoTrackSelection;
+import androidx.media3.exoplayer.upstream.CmcdConfiguration;
+import androidx.media3.exoplayer.upstream.LoaderErrorThrower;
+
+import java.util.List;
 
 public class DefaultDashChunkSource {
-    public static final class Factory {
+    public static final class Factory implements DashChunkSource.Factory {
```
Comment (freeboub) [inline android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java]: I don't understand the changes in this file... Did you try to build without dash media3 dash plug in?

**Review by freeboub — CHANGES_REQUESTED** [review]

File: android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java (hunk 2 above)
Comment (seyedmostafahasani) [reply android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java]: After converting `DashMediaSource` from Java to Kotlin, I encountered an error in `ReactExoPlayerView`.
```
mediaSourceFactory = new DashMediaSource.Factory(
                        new DefaultDashChunkSource.Factory(mediaDataSourceFactory),
                        buildDataSourceFactory(false)
                );
```
We need to add the following changes.

Comment (freeboub) [reply android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java]: I think:
import androidx.media3.exoplayer.dash.manifest.DashManifest;
Will fail if you disable dash support, can you please triple check ?

Comment (seyedmostafahasani) [reply android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java]: For sure, I will test it.

Comment (seyedmostafahasani) [reply android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java]: @freeboub 
You're right. 
I reverted the code.

**Review by freeboub — APPROVED** [review]

File: android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java (hunk 2 above)
Comment (freeboub) [reply android/src/main/java/androidx/media3/exoplayer/dash/DefaultDashChunkSource.java]: Great, thanks!

**Comment by seyedmostafahasani** [comment]
I found a new problem with the `mpd` format. Please don't merge this PR yet. I am trying to find what the problem is.

**Pull request closed without merging**